### PR TITLE
feat: Support full URLs when adding repo to insights page

### LINKED ISSUE #1415: Feature: Allow use of full repository URL when adding a repository to an insights page
### Type of feature

🍕 Feature

### Current behavior

If I try to copy/paste a repository URL from github.com into the field for adding it to my insights page, I get an error message. I am instead required to use only the part of the URL that is repository's full name.

2023-07-24 19 29 03 insights opensauced pizza f17067e3cb76

### Suggested solution

I would like to be able to just enter the full URL in this field. This could probably be achieved by having a script on the frontend that recognizes Github URLs and can pull from it the information that the field requires.

I'd be happy to work on this if this repository is looking for contributions from the community.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by dominicduffin1**
## Description

This PR adds support for full URLs to the Add Repository selection field when creating/updating an insights page.

I've used a code solution from this repo: [https://github.com/open-sauced/open-sauced/blob/main/src/lib/util.js#L1](https://github.com/open-sauced/open-sauced/blob/main/src/lib/util.js#L1) and added tests.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #1415 

## Mobile & Desktop Screenshots/Recordings

I haven't changed any of the language here, but perhaps I should?

2023-08-02 19 50 31 localhost 55008dd89a7a

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks @dominicduffin1. We have a util function already for validating github URLs here https://github.com/open-sauced/insights/blob/beta/lib/utils/github.ts#L24. Will you use that instead?

**Comment by dominicduffin1** [comment]
@brandonroberts Yes I can revise to use that.

**Review by babblebey — COMMENTED** [review]
Hey dominic, quick reminder to update the integrated `Search` component `placeholder` prop value currently

```js
<Search
  placeholder="Repository Full Name (ex: open-sauced/open-sauced)"
  // ...
/>
```

Update to something that notifies users of the full Repo URL support...

**Comment by babblebey** [comment]
The placeholder being long isn't something we can avoid, the message still has to be passed out as correctly as possible IMHO 😉

That part LGTM 🍕

**Comment by brandonroberts** [comment]
@dominicduffin1 looks better. Instead of copying the other function and tweaking the return, how about we give the original utility function with a more generic name and use it instead, such as `generateRepoParts` or something similar.

**Comment by dominicduffin1** [comment]
@brandonroberts We certainly can do that, looking in more detail at how it's been used I think it should be safe to add the logic this solution needs to the original one.

**Comment by brandonroberts** [comment]
> @brandonroberts We certainly can do that, looking in more detail at how it's been used I think it should be safe to add the logic this solution needs to the original one.

Ok cool

**Comment by dominicduffin1** [comment]
I've just resolved the merge conflicts from the latest version of `beta` and pushed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,13 +7,12 @@
 
 ## [1.60.0-beta.6](https://github.com/open-sauced/insights/compare/v1.60.0-beta.5...v1.60.0-beta.6) (2023-08-11)
 
-
```
Comment (brandonroberts) [inline CHANGELOG.md:10]: Revert changes to this file

**Review by brandonroberts — APPROVED** [review]
Code LGTM. Minor revert

**Comment by brandonroberts** [comment]
Thanks @dominicduffin1. After the changelog is reverted and the merge conflicts are resolved, we should be good to go

**Comment by dominicduffin1** [comment]
I've resolved the latest merge conflicts and reverted the `CHANGELOG.md` change so we should now have a clean PR.

**Pull request merged**